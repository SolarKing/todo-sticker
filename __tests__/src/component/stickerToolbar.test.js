// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import StickerToolBar from '../../../src/component/stickerToolbar';
import stickerDispatcher from '../../../src/sticker/dispatcher';

describe('StickerToolbar', () => {
    let wrapper: ShallowWrapper;

    beforeAll(() => {
        wrapper = shallow(<StickerToolBar/>);
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should dispatch minimize window action', () => {
        let mockDispatch = jest.spyOn(stickerDispatcher, 'dispatch');
        wrapper.find('ToolButton').at(0).simulate('click');
        expect(mockDispatch.mock.calls.length).toEqual(1);
        expect(mockDispatch.mock.calls[0][0]).toEqual({channel: 'minimizeStickerWindow'});
    });

    it('should dispatch maximize window action', () => {
        let mockDispatch = jest.spyOn(stickerDispatcher, 'dispatch');
        wrapper.find('ToolButton').at(1).simulate('click');
        expect(mockDispatch.mock.calls.length).toEqual(1);
        expect(mockDispatch.mock.calls[0][0]).toEqual({channel: 'maximizeStickerWindow'});
    });

    it('should dispatch close window action', () => {
        let mockDispatch = jest.spyOn(stickerDispatcher, 'dispatch');
        wrapper.find('ToolButton').at(2).simulate('click');
        expect(mockDispatch.mock.calls.length).toEqual(1);
        expect(mockDispatch.mock.calls[0][0]).toEqual({channel: 'closeStickerWindow'});
    });

});