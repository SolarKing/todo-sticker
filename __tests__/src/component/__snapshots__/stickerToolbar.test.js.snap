// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickerToolbar should render correct 1`] = `
<stickerToolbar__Container>
  <ToolButton
    className="far fa-window-minimize"
    onClick={[Function]}
  />
  <ToolButton
    className="far fa-window-maximize"
    onClick={[Function]}
  />
  <ToolButton
    className="far fa-window-close"
    onClick={[Function]}
  />
</stickerToolbar__Container>
`;
